src/controllers/timeTaskCtrl.js

```diff
@@ -0,0 +1,173 @@
+          .catch((error) => res.status(500).json({ error }));
+      } else {
+        let lastTime = data.timeTask.time[data.timeTask.time.length - 1];
+        // If end_date is null so a timer is running
+        if (!(lastTime.end_date)) {
+          return res.status(400).json({ message: 'A timer is running on this task, end the timer and retry' })
+        }
+
+        // Else init a new timer
+        const updTime = {
+          time: [
+            ...data.timeTask.time,
+            {
+              start_date: new Date().toISOString()
+            }
+          ]
+        }
+
+        // Update TimeTask with the new timer
+        TimeTask.findByIdAndUpdate(data.timeTask.id, updTime, { new: true })
+          .then((result) => {
+            if (!result) {
+              return res.status(404).json({ message: 'Timer not found' });
+            }
+            return res
+              .status(200)
+              .json({ message: 'The timer has been modified correclty', result });
+          })
+          .catch((error) => res.status(400).json({ error }));
+      }
+    })
+    .catch(error => res.status(400).json({ error }))
+};
+
+exports.endTimeTask = (req, res) => {
+  getTimeTask(req.params.task_id)
+    .then((data) => {
+      // data.message mean timeTask not found, so return error
+      if (data.message) {
+        return res.status(404).json({ message: 'Timer not found, start a new timer for init' });
+      }
+
+      let lastTime = data.timeTask.time[data.timeTask.time.length - 1];
+      // If end_date is true, no timer is running and return error message
+      if (lastTime.end_date) {
+        return res.status(400).json({ message: 'No timer started' })
+      }
+
+      // Else stop current timer is running
+      lastTime.end_date = new Date().toISOString();
+      let updTime = {
+        time: [
+          ...data.timeTask.time
+        ]
+      }
+
+      TimeTask.findByIdAndUpdate(data.timeTask.id, updTime, { new: true })
+        .then((result) => {
+          if (!result) {
+            return res.status(404).json({ message: 'Timer not found' });
+          }
+          return res
+            .status(200)
+            .json({ message: 'The timer has been modified correclty', result });
+        })
+        .catch((error) => res.status(400).json({ error }));
+    })
+    .catch((error) => res.status(400).json({ error }));
+};
+
+exports.getTimeTasks = (_, res) => {
+  TimeTask.find()
+    .then((timeTasks) => res.status(200).json({ timeTasks }))
+    .catch((error) => res.status(400).json({ error }));
+};
+
+exports.getOneTimeTask = (req, res) => {
+  getTimeTask(req.params.task_id)
+    .then((data) => {
+      if (data.message) {
+        return res.status(404).json({ message: data.message });
+      }
+      return res.status(200).json({ timeTask: data.timeTask });
+    })
+    .catch((error) => res.status(400).json({ error }));
+};
```

Comment:
Could you create a route that directly returns the current running time task of user?

Reply:
Seems legit, for get current task when user reconnect on app.
And no problem, all upgrade is welcome.